### PR TITLE
Update HPCS module with ibm_hpcs resource

### PR DESCRIPTION
**Pull request by kavya498**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,13 @@
-# Managing HPCS Service Instances using Terraform Modules
```
Comment (umarali-nagoor) [inline README.md:1]: Add detect secret part in github actions (https://github.com/kavya498/terraform-ibm-hpcs/blob/hpcs/.github/workflows/validate_terraform.yml)

and add even in .pre-commit.yaml file

generate .secret.base file and check-in

take observability module as a reference

File: examples/ibm-hpcs/variables.tf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+variable "service_name" {
```
Comment (umarali-nagoor) [inline examples/ibm-hpcs/variables.tf:1]: Sorry just want to know why we are allowing user to pass only following 4 parameters (service_name, admins, key_name and region) values, rest all parameters given some default values on module side

This template allow user to input only above 4 parameters, can we add more input parameters

File: README.md [hunk 3]
```diff
@@ -103,73 +48,36 @@ module "ibm-hpcs-kms-key" {
 
 | Name              | Description                                                                    | Type   |Default  | Required|
```
Comment (umarali-nagoor) [inline README.md:49]: Capture the outputs as well

File: outputs.tf [hunk 4]
```diff
@@ -0,0 +1,18 @@
+output "hpcs_guid" {
+  value = local.instance_id
+}
+output "key_id" {
+  value = ibm_kms_key.key[0].key_id
+}
+output "key_crn" {
+  value = ibm_kms_key.key[0].id
+}
+output "key_alias" {
```
Comment (umarali-nagoor) [inline outputs.tf:10]: Based on below condition,  `ibm_kms_key_alias` may not get provisioned when user passes null value

count         = var.key_alias != null ? 1 : 0

in such cases add condition check before trying to fetch the output value

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+variable "expiration_date" {
+  description = "Expiration_date of the Key"
+  type        = string
+  default     = null
+}
+
+######################################
+# HPCS Key Policy Variables
+######################################
+variable "rotation" {
```
Comment (umarali-nagoor) [inline variables.tf:127]: Can we define the contents of rotation object like you are searching for "interval_month" in it. Just like you added

variable "admins" {
  type = list(object({
    name  = string
    key   = string
    token = string
  }))
  description = "The list of administrators for the instance crypto units. "
}

File: variables.tf [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+  description = "Rotaion policy"
+  type        = any
+  default     = []
+}
+variable "dual_auth_delete" {
```
Comment (umarali-nagoor) [inline variables.tf:132]: Please avoid using type "any" as much as possible and define the contents of object.

One of the feedback got from users as they are not sure what to mention when the type is "any"

File: variables.tf [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+  description = "Dual auth policy"
+  type        = any
+  default     = []
```
Comment (umarali-nagoor) [inline variables.tf:135]: same as above

File: variables.tf [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+variable "failover_units" {
+  type        = number
+  description = "Number of failover units"
+  default     = null
+}
+variable "location" {
+  default     = "us-south"
+  type        = string
+  description = "Location of HPCS Instance. Allowed values are us-south and us-east"
+  validation {
```
Comment (umarali-nagoor) [inline variables.tf:19]: We can use validation check as follows

variable "hardware" {
  description = "The level of hardware isolation for your worker node."
  type        = string
  validation  {
      error_message = "Level of hardware isolation, Allowed values are `shared` and `dedicated`."
      condition     = can(
        contains([
          "shared",
          "dedicated"
        ], var.hardware)
      )
    }
}

File: variables.tf [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+    condition     = var.location == "us-south" || var.location == "us-east"
+    error_message = "Hpcs initialisation is only supported in us-south and us-east via terraform."
+  }
+}
+variable "service_name" {
```
Comment (umarali-nagoor) [inline variables.tf:24]: Please add regex validation, e.g:

variable "cluster_name" {
  description = "Name of the cluster"
  type        = string
  validation  {
      error_message = "Cluster name must begin and end with a letter and contain only letters, numbers, and - characters."
      condition     = can(regex("^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$", var.cluster_name))
    }
}

**Comment by daniel-butler-irl** [comment]
stale PR closing

**Pull request closed without merging**